src/runtime/task/mod.rs

```diff
@@ -53,34 +59,49 @@ pub(crate) struct Task {
 
 impl Task {
     /// Create a task from a continuation
-    fn new<F>(f: F, stack_size: usize, id: TaskId, task_type: TaskType, name: Option<String>) -> Self
+    fn new<F>(
+        f: F,
+        stack_size: usize,
+        id: TaskId,
+        task_type: TaskType,
+        name: Option<String>,
+        clock: VectorClock,
+    ) -> Self
     where
         F: FnOnce() + Send + 'static,
     {
         let mut continuation = ContinuationPool::acquire(stack_size);
         continuation.initialize(Box::new(f));
         let waker = make_waker(id);
         let continuation = Rc::new(RefCell::new(continuation));
+        assert!(id.0 < clock.time.len());
```

Comment:
nit: probably move this assert to the top of the method body

Reply:
Done.